cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -174,9 +175,16 @@ protected void configure() {
                                                             AlertPublisher.PLUGIN_TYPE),
                                             config.getEngine()).generateSpec(config);
     } catch (ValidationException e) {
-      throw new IllegalArgumentException(
-        String.format("Failed to configure pipeline: %s",
-                      e.getFailures().isEmpty() ? e.getMessage() : e.getFailures().iterator().next().getMessage()), e);
+      String errorMessage = e.getMessage();
+      if (!e.getFailures().isEmpty()) {
+        ValidationFailure failure = e.getFailures().iterator().next();
+        if (failure.getCorrectiveAction() != null) {
```

Comment:
seems like it would be useful to put this in the ValidationFailure class, perhaps as a getFullMessage() method or something like that.

Reply:
yea. the pattern was repetitive. Fixed.